plugin/core/windows.py

```diff
@@ -128,6 +128,12 @@ def disable_config_async(self, config_name: str) -> None:
         self._config_manager.disable_config(config_name)
 
     def register_listener_async(self, listener: AbstractViewListener) -> None:
+        config = self._needed_config(listener.view)
+        if config:
+            plugin = get_plugin(config.name)
+            if plugin and plugin.should_ignore(listener.view):
+                debug(listener.view, "ignored by plugin", plugin.__name__)
+                return
```

Comment:
I don't know if this part specifically but it can get into an infinite loop of starting and closing the server. I think when the first opened file is ignored.

Reply:
Do you have an example where you have seen such an infinite loop? It seems to work as expected on my side, even when the first opened file is ignored.

Reply:
It seems to be a bit random. I've modified LSP-pyright to ignore all files in its `plugin.py` and then restarted LSP.
Also note that I have 3 different servers starting on that file (LSP-pylsp, LSP-ruff and LSP-pyright).

Reply:
I could reproduce it with multiple sessions. This code was indeed misplaced here, because it would disable the listener entirely if any of the applicable sessions ignores the view. I don't know why this did lead to the infinite starting loop, but it's somewhat difficult to understand (for me) what exactly happens in the `_dequeue_listener_async` method, which calls itself recursively. I think the check for ignored views is in the right place now, and this bug should be fixed now.

Reply:
I still see this problem.

After adding this to LSP-pyright:

```py
    @classmethod
    def should_ignore(cls, view: sublime.View) -> bool:
        return True
```

None of the sessions are starting anymore on `LSP-pyright/plugin.py`. I'm expecting LSP-pylsp and LSP-ruff to still be able to start.

Reply:
It seemed to work randomly for me, sometimes it worked and sometimes it didn't. Could you try again with the latest changes?